### PR TITLE
Add defaults for common dask-cuda cluster configurations (tcp, ucx+noIB, ucx+IB)

### LINKED ISSUE #1: [FEA] Add cluster setup scripts for different communication protocols/setups
An update to the existing scripts would to support different protocol options and utilize the different networking hardware present in a given cluster. For the time being I see 3 options:
- TCP only
- UCX + NVLink + no-ib
- UCX + NVLink + IB

This could either be separated out into 3 different scripts that export relevant variables and build the appropriate scheduler and worker arguments, or be implemented as different function within the existing cluster startup scripts with an additional parameter defining the cluster type. 

Do others have a preference on which method they prefer? 

Happy to take this on make updates. 

cc: @rlratzel

### PR DESCRIPTION
**Pull request by ayushdg**
Fixes #1 

- [X] Update `run-dask-process.sh` to accept an additional parameter defining the cluster type.
- [x] Update `wait-for-workers.py` to accept cluster type

**Note**: This updates the current behavior of the script where if not explicitly mentioned, the cluster type will be TCP

### CODE REVIEW

File: run-dask-process.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ ARGS=$*
 function hasArg {
     (( ${NUMARGS} != 0 )) && (echo " ${ARGS} " | grep -q " $1 ")
 }
-VALIDARGS="-h --help scheduler workers"
+VALIDARGS="-h --help scheduler workers tcp ucx ucxib ucx-ib"
```
Comment (ayushdg) [inline run-dask-process.sh:29]: Was looking at a better way to implement this. Something like a `--mode/cluster-type tcp` but went down a weird rabbit hole of `getopt` vs `getopts`. Open to suggestions/recommendations here

File: run-dask-process.sh [hunk 2]
```diff
@@ -65,47 +65,104 @@ fi
...
-                --interface=ib0
-                --scheduler-file $SCHEDULER_FILE
-               "
 
-WORKER_ARGS="--enable-tcp-over-ucx
-             --enable-nvlink 
-             --enable-infiniband
-             --enable-rdmacm
-             --rmm-pool-size=$WORKER_RMM_POOL_SIZE
+DASK_SCHEDULER_PORT=${DASK_SCHEDULER_PORT:-8792}
```
Comment (ayushdg) [inline run-dask-process.sh:73]: Potentially move this to the `default-config.sh` script

File: run-dask-process.sh [hunk 3]
```diff
@@ -65,47 +65,104 @@ fi
...
+    export UCX_MEMTYPE_REG_WHOLE_ALLOC_TYPES=cuda
+
+    export DASK_UCX__CUDA_COPY=True
+    export DASK_UCX__TCP=True
+    export DASK_UCX__NVLINK=True
+    export DASK_UCX__INFINIBAND=True
+    export DASK_UCX__RDMACM=True
+    export DASK_RMM__POOL_SIZE=0.5GB
+
+    UCXPY_INTERFACE=${UCXPY_INTERFACE:-ib0}
```
Comment (ayushdg) [inline run-dask-process.sh:109]: Same as above

File: run-dask-process.sh (hunk 1 above)
Comment (rlratzel) [reply run-dask-process.sh:29]: yeah, I think the move to start using `getopt`/`getopts` is long overdue.  I'm in favor of either, but I'd need more time to look into each to see if there's pros and cons that apply.

File: run-dask-process.sh (hunk 2 above)
Comment (rlratzel) [reply run-dask-process.sh:73]: This is a good idea.

File: run-dask-process.sh (hunk 3 above)
Comment (rlratzel) [reply run-dask-process.sh:109]: 👍

**Review by rlratzel — APPROVED** [review]
LGTM. I noticed you added the "breaking" label, but it seems like it's backwards-compatible by default, or did I miss the breaking changes?

File: run-dask-process.sh [hunk 4]
```diff
@@ -26,11 +26,12 @@ ARGS=$*
...
 }
-VALIDARGS="-h --help scheduler workers"
+VALIDARGS="-h --help scheduler workers tcp ucx ucxib ucx-ib"
 HELP="$0 [<app> ...] [<flag> ...]
  where <app> is:
-   scheduler        - start dask scheduler
-   workers          - start dask workers
+   scheduler               - start dask scheduler
+   workers                 - start dask workers
+   tcp/ucx/ucxib/ucx-ib    - communication type for cluster
```
Comment (rlratzel) [inline run-dask-process.sh:34]: This isn't required but I've often followed the convention of "options" vs. "targets" for these, where options start with `--`.  For this, you could say `--tcp`, etc. are all options for how to start the workers and scheduler targets.

**Review by rlratzel — APPROVED** [review]
This is great, thanks!  I had one minor suggestion which is more of a preference and need not hold up approval.

File: run-dask-process.sh (hunk 4 above)
Comment (ayushdg) [reply run-dask-process.sh]: Agreed. Updated to have these as flags rather than targets. Let me know if the change look good and I'll merge.

**Review by rlratzel — APPROVED** [review]
LGTM, thanks!

**Pull request merged**